packages/medusa/src/loaders/database.ts

```diff
@@ -1,48 +1,61 @@
 import {
-  Connection,
-  createConnection,
-  getConnectionManager,
-  getConnection,
+  DataSource,
+  DataSourceOptions,
+  Repository,
+  TreeRepository,
 } from "typeorm"
-import { ShortenedNamingStrategy } from "../utils/naming-strategy"
 import { AwilixContainer } from "awilix"
-import { ConnectionOptions } from "typeorm/connection/ConnectionOptions"
 import { ConfigModule } from "../types/global"
+import "../utils/naming-strategy"
 
 type Options = {
   configModule: ConfigModule
   container: AwilixContainer
+  customOptions?: {
+    migrations: DataSourceOptions["migrations"]
+    logging: DataSourceOptions["logging"]
+  }
+}
+
+export let dataSource: DataSource
+
+if (process.env.NODE_ENV === "test") {
+  dataSource = {
+    getRepository: (target) => new Repository(target, {} as any) as any,
+    getTreeRepository: (target) => new TreeRepository(target, {} as any) as any,
+  } as unknown as DataSource
 }
 
 export default async ({
   container,
   configModule,
-}: Options): Promise<Connection> => {
+  customOptions,
+}: Options): Promise<DataSource> => {
   const entities = container.resolve("db_entities")
 
   const isSqlite = configModule.projectConfig.database_type === "sqlite"
 
-  const cnnManager = getConnectionManager()
-  if (cnnManager.has("default") && getConnection().isConnected) {
-    await getConnection().close()
-  }
-
-  const connection = await createConnection({
+  dataSource = new DataSource({
     type: configModule.projectConfig.database_type,
     url: configModule.projectConfig.database_url,
     database: configModule.projectConfig.database_database,
     extra: configModule.projectConfig.database_extra || {},
     schema: configModule.projectConfig.database_schema,
     entities,
-    namingStrategy: new ShortenedNamingStrategy(),
-    logging: configModule.projectConfig.database_logging || false,
-  } as ConnectionOptions)
+    migrations: customOptions?.migrations,
+    // namingStrategy: new DefaultNamingStrategy(), Since we are monkey patching, no need to set this option
```

Comment:
question: Should we go ahead and remove it then, or is there a reason to keep it in (other than for explanatory purposes)?